base/replicator.go

```diff
@@ -242,3 +243,15 @@ func (r *Replicator) populateActiveTaskFromReplication(replication sgreplicate.S
 
 	return
 }
+
+func (r *Replicator) StopReplications() error {
+	 r.lock.Lock()
+	 defer r.lock.Unlock()
+	for replicationId, replication := range r.replications {
+		LogTo("Replicate", "Stopping replication %s", replicationId)
+		if err := replication.Stop(); err != nil {
```

Comment:
After seeing the locking change, I wanted to understand what exactly this was doing.

Why is this calling replication.Stop(), and not r.stopReplication(repId)?  It looks like r.stopReplication also does things (updating active tasks, and removing the replication from r.replications).

Reply:
Thanks for catching!  I didn't notice `r.stopReplication(repId)` -- I've made the changes and kicked off a new build + integration tests